cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -562,7 +562,12 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         predicateValue = negatedPredicate.get(operator);
                         placeholder = "?";
-                        sqlParameters.add(String.valueOf(predicateValue));
+                        if (predicateValue instanceof Boolean){
```

Comment:
If we are going to do that, and remove the `String.valueOf` calls, we should push it as a single task for a refactor rather than doing it here, since we'd then be calling `String.valueOf(o)` multiple times.I'm not sure why we'd need to be mindful of the `Integer` types since wouldn't they drop into the else and use `String.valueof` anyway?